src/keys.rs

```diff
@@ -1198,6 +1189,22 @@ mod tests {
         ));
     }
 
+    #[test]
+    fn issuance_authorizing_key_from_bytes_fail_on_zero() {
+        // isk must not be the zero scalar.
+        let isk = IssuanceAuthorizingKey::from_bytes([0; 32]);
+        assert!(isk.is_none());
+    }
+
+    #[test]
+    fn issuance_authorizing_key_from_bytes_to_bytes_roundtrip() {
+        let isk = IssuanceAuthorizingKey::random();
+        let isk_bytes = isk.to_bytes();
+        let isk_roundtrip = IssuanceAuthorizingKey::from_bytes(isk_bytes);
+        assert!(isk_roundtrip.is_some());
+        assert_eq!(isk.to_bytes(), isk_roundtrip.unwrap().to_bytes());
```

Comment:
The PartialEq and Eq traits cannot be derived for IssuanceAuthorizingKey since they are not implemented for NonZeroScalar, so I check that the byte representations match, which seems reasonable.

Reply:
That looks much nicer, thanks